### DISCUSSION TITLE
Which version of TLS does this sdk use?

### DISCUSSION THREAD

**Discussion by coderavels** [microsoftgraph/msgraph-sdk-go#666] [Q&A]
Saw this warning on the [download file](https://learn.microsoft.com/en-us/graph/api/driveitem-get-content?view=graph-rest-1.0&tabs=http) microsoft document
```
 Warning

This endpoint requires TLS 1.2 to function normally. Microsoft announced the deprecation of TLS 1.0 and 1.1 for Office 365 and Microsoft Entra services. Although Microsoft Graph still supports these two protocols, you might experience transport-level errors. For more information about the TLS 1.0 and 1.1 deprecation, see Enable support for TLS 1.2 in your environment.
```

Wanted to confirm if the sdk uses TLS 1.2 and we don't need to take any extra precautions for this warning.

**Comment by baywet** [comment]
This SDK relies on net/http, which in turns relies on crypto/tls. [Crypto/tls](https://pkg.go.dev/crypto/tls) supports TLS1.2 and 1.3.

In fact I believe this documentation page is outdated since [Microsoft Graph has disabled TLS 1.0/1.1](https://www.ssllabs.com/ssltest/analyze.html?d=graph.microsoft.com&s=2603%3a1037%3a1%3a110%3a0%3a0%3a0%3a80) for security reasons, and only serves requests with 1.2/1.3
